Automattic/woocommerce-payments
# Display Link as payment method on order received page with setup intents

Fixes #8122 

#### Changes proposed in this Pull Request

This PR ensures that Link, rather than Credit / Debit Card, is displayed as the payment method on the order received page for Link orders using setup intents. The problem is that the payment method type, `card` or `link`, is pulled from the `payment_method_details` object within the intent. Setup intents don't contain `payment_method_details` so the payment method type is pulled from `payment_method_options`, which only contains `card`. To get around this, we can look at the token and determine if it's a Link token and set the payment method type accordingly.









#### Testing instructions





This specifically deals with setup intents so assume all orders should be setup intents, such as a subscription with a free trial.

Repeat the following steps on Shortcode and Blocks checkouts:

* Place an order using card(no Link) as the payment method and check that the Order Received page shows Credit / Debit Card as the payment method
* Place an order using Link as the payment method with a normal card and check that the Order Received page shows Link as the payment method
* Place an order using Link as the payment method with a 3DS card(the **Always authenticate** card [here](https://docs.stripe.com/testing?testing-method=card-numbers#regulatory-cards) works well) and check that the Order Received page shows Link as the payment method

-------------------


**Post merge**
